### [CVE-2020-5235](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-5235)
![](https://img.shields.io/static/v1?label=Product&message=Nanopb&color=blue)
![](https://img.shields.io/static/v1?label=Version&message=%3C%200.2.9.4%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Version&message=%3E%3D%200.3.0%2C%20%3C%200.3.9.5%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Version&message=%3E%3D%200.4.0%2C%20%3C%200.4.1%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Vulnerability&message=CWE-125%3A%20Out-of-bounds%20Read&color=brightgreen)

### Description

There is a potentially exploitable out of memory condition In Nanopb before 0.4.1, 0.3.9.5, and 0.2.9.4. When nanopb is compiled with PB_ENABLE_MALLOC, the message to be decoded contains a repeated string, bytes or message field and realloc() runs out of memory when expanding the array nanopb can end up calling `free()` on a pointer value that comes from uninitialized memory. Depending on platform this can result in a crash or further memory corruption, which may be exploitable in some cases. This problem is fixed in nanopb-0.4.1, nanopb-0.3.9.5, nanopb-0.2.9.4.

### POC

#### Reference
No PoCs from references.

#### Github
- https://github.com/t-heGhos-t/t-heGhos-t

